yt_dlp/extractor/curiositystream.py

```diff
@@ -23,9 +23,11 @@ def _call_api(self, path, video_id, query=None):
             auth_cookie = self._get_cookies('https://curiositystream.com').get('auth_token')
             if auth_cookie:
                 self.write_debug('Obtained auth_token cookie')
-                self._auth_token = auth_cookie.value
+                self._auth_token = compat_urllib_parse_unquote(auth_cookie.value)
+            else:
+                self.write_debug('Auth cookie NOT found')
```

Comment:
unnecessary

Reply:
It helped with debugging. This is not a first time this broke. Ok, I'll remove it. Personally, I would prefer some command line argument to tell yt-dlp to crash if auth fails. It is annoying start watching a video and then have it cut at 2 minutes, even though download was completed successfully.

Reply:
If login fails (`_perform_login`), error is raised, no?

Reply:
> It helped with debugging. 

The `if` block has a debug message, making this redundant

Reply:
> If login fails (`_perform_login`), error is raised, no?

It think it doesn't, when using cookies from browser for auth (it just doesn't send the header). But you are right, I should try using the auth arguments instead of taking it from the browser (it is more convenient to use cookies for me, because I am always logged-in in the browser; didn't have to complicate my custom download script to handle auth for different services).